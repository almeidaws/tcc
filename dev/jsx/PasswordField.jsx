import Field from './Field.jsx';
import React from 'react';

const PasswordField = props => <Field 
                                    type="password" 
                                    placeholder="Enter Password" 
                                    icon="lock" 
                                    required={true}
                                    errorMessage={props.errorMessage}
                                    value={props.value} 
                                    onChange={props.onChange} />;
export default PasswordField;