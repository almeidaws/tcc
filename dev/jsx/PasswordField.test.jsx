const React = require('react');
import PasswordField from './PasswordField.jsx';
const renderer = require('react-test-renderer');

it('<PasswordField /> should render password', () => {
    const component = renderer.create(<PasswordField
                                        value="myPassword"
                                        />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('<PasswordField /> should presenter an error', () => {
    const component = renderer.create(<PasswordField
                                        value="myPassword"
                                        />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});