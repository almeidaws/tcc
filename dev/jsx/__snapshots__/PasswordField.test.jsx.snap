// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordField /> should presenter an error 1`] = `
<div
  className="form-group"
>
  <input
    className="form-control"
    placeholder="Enter Password"
    required={true}
    type="password"
    value="myPassword"
  />
  <span
    className="form_icon"
  >
    <i
      aria-hidden="true"
      className="fa_icon form-lock"
    />
  </span>
  <div
    className="invalid-feedback text-warning"
  />
</div>
`;

exports[`<PasswordField /> should render password 1`] = `
<div
  className="form-group"
>
  <input
    className="form-control"
    placeholder="Enter Password"
    required={true}
    type="password"
    value="myPassword"
  />
  <span
    className="form_icon"
  >
    <i
      aria-hidden="true"
      className="fa_icon form-lock"
    />
  </span>
  <div
    className="invalid-feedback text-warning"
  />
</div>
`;
